source/credentials_utils.c

```diff
@@ -189,26 +190,39 @@ struct aws_credentials *aws_parse_credentials_from_aws_json_object(
         goto done;
     }
 
-    struct aws_byte_cursor secrete_access_key_cursor;
-    secrete_access_key = aws_json_value_get_from_object(
+    struct aws_byte_cursor secret_access_key_cursor;
+    secret_access_key = aws_json_value_get_from_object(
         document_root, aws_byte_cursor_from_c_str((char *)options->secret_access_key_name));
-    if (!aws_json_value_is_string(secrete_access_key) ||
-        aws_json_value_get_string(secrete_access_key, &secrete_access_key_cursor) == AWS_OP_ERR) {
+    if (!aws_json_value_is_string(secret_access_key) ||
+        aws_json_value_get_string(secret_access_key, &secret_access_key_cursor) == AWS_OP_ERR) {
         AWS_LOGF_ERROR(AWS_LS_AUTH_CREDENTIALS_PROVIDER, "Failed to parse SecretAccessKey from Json document.");
         goto done;
     }
 
     struct aws_byte_cursor token_cursor;
+    AWS_ZERO_STRUCT(token_cursor);
     if (options->token_name) {
         token = aws_json_value_get_from_object(document_root, aws_byte_cursor_from_c_str((char *)options->token_name));
-        if (!aws_json_value_is_string(token) || aws_json_value_get_string(token, &token_cursor) == AWS_OP_ERR) {
+        if (!aws_json_value_is_string(token) || aws_json_value_get_string(token, &token_cursor) == AWS_OP_ERR ||
+            token_cursor.len == 0) {
```

Comment:
why is this one getting a length check, but `access_key_id`  and `secret_access_key` above don't get a length check?

worth breaking these 4 checks out into a helper function?

Reply:
`access_key_id` and `secret_access_key` are required parameters for credentials, so we will fail to create the credentials struct if they are empty. So we don't need to do any length check here. `aws_session_token` is an optional field, but this parser has an option of `token_required`, if that is set, we want to error out if it's empty.